python/cudf/cudf/core/buffer/buffer.py

```diff
@@ -171,6 +172,37 @@ def __getitem__(self, key: slice) -> Buffer:
             raise ValueError("slice must be C-contiguous")
         return self._getitem(offset=start, size=stop - start)
 
+    def copy(self, deep: bool = True):
```

Comment:
cc: @madsbk 

I realize that spilling and copy-on-write cannot be currently used together, but we still want to make sure this PR does not break spilling.

Does `SpillableBuffer` need to implement a `copy()` method or can it just safely inherit `Buffer.copy`?

Reply:
Resolved by @madsbk suggestion below.